app/src/Event/EventController.php

```diff
@@ -1149,6 +1150,74 @@ public function editTracks($friendly_name)
         );
     }
 
+    /**
+     * Upload Data from CSV for this event
+     * @todo Validate & Process uploaded cSV
+     * @param string $friendly_name
+     */
+    public function eventImportCsv($eventSlug)
+    {
+        $config = $this->application->config('oauth');
+        $request = $this->application->request();
+
+        /** @var FormFactoryInterface $factory */
+        $factory = $this->application->formFactory;
+        $form    = $factory->create(new EventImportFormType());
+
+        //use EventFormImportType to validate the CSV is valid
+        //We possibly want to do some extensive data checking
+        //BEFORE we blindly throw data from here to the API
+        if ($request->isPost()) {
+            try {
+                if (isset($_FILES['event_import']['error']['csv_file'])
+                    && $_FILES['event_import']['error']['csv_file'] == UPLOAD_ERR_OK) {
+                    $eventApi = $this->getEventApi();
+                    $event = $eventApi->getByFriendlyUrl($eventSlug);
+                    $handle = fopen($_FILES['event_import']['tmp_name']['csv_file'], "r");
+
+                    while (!feof($handle)) {
+                        $talk = fgetcsv($handle);
+                        $date_start = new \DateTime();
+                        $date_start->setTimestamp(
+                            strtotime($talk[3].' '.$talk[4])
+                        );
+                        $date_end = new \DateTime();
+                        $date_end->setTimestamp(
+                            strtotime($date_start->format($date_end::ATOM).' plus '.$talk[5].' minutes')
```

Comment:
Why the time-magic here? Assuming that the CSV-File contains the fields "date", "starttime" and "duration" you could also do this:

```php
$date_start = new DateTimeImmutable($talk[3] . ' ' . $talk[4]);
$date_end = $date_start->add(new DateInterval('PT' . $talk[5] . 'M'));
```

Additionally I'm missing the Timezone-Information. As far as I can tell this will then be imported using the local timezone which might lead to unexpected results. 

So perhaps something like this would help here:

```php
$date_start = new DateTimeImmutable($talk[3] . ' ' . $talk[4], $event->getFullTimezone());
```

Reply:
The timezone seems to be irrelevant here as the DateTimeInterface will be "reduced" to `Y-m-d H:i` almost immediately in `TalkApi::addTalk`. Still it might help with reducing errors due to wrongly set default-timezone.

Reply:
Like you stated, all the times are being handled by the Talkapi endpoint and methods. As long as my CSV has the talks in the event's timezone it'll work out as expected.

Reply:
It'll not necessarily work out as expected. But that will only happen in so weird special edge-cases that I'm with you here that the timezone-information can be skipped here.

Should weird things happen they need to be addressed later via the web-UI